### PR TITLE
Provide immutable system accounts `0.0.800` and `0.0.801` for staking funding

### LINKED ISSUE #3015: Enable immutable system accounts
### Problem

At present, we require all non-contract accounts to have at least one associated cryptographic key. 

Trying to create or update an account with the empty `KeyList` (the sentinel key for other immutable entities) results in `BAD_ENCODING`.

These semantics are correct for non-system accounts. (That is, accounts that can be created via HAPI `CryptoCreate`.)  But it may also be useful to have a system account---that is, an account below `0.0.1000`, not created via HAPI---that can receive ℏ, but never transfer it.

### Solution

- Allow non-contract accounts to signal immutability via an empty `KeyList` (as with other entity types).
- Update the [relevant `SigMetaLookup` method](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/sigs/metadata/StateChildrenSigMetadataLookup.java#L224) to return a new `IMMUTABLE_ACCOUNT` lookup failure for any account whose key is the empty `KeyList`. (In 0.25 we can add a new HAPI response code so that this signing failure can be externalized in receipts and records.)

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
- Introduces [a `MigrationRecordsManager`](https://github.com/hashgraph/hedera-services/blob/03015-M-ImmutableSystemAccounts/hedera-node/src/main/java/com/hedera/services/state/migration/MigrationRecordsManager.java) responsible for streaming child records that give mirror nodes visibility into any otherwise "hidden" state changes that happened during migration.
- Adds a release 0.24 [migration](https://github.com/hashgraph/hedera-services/blob/03015-M-ImmutableSystemAccounts/hedera-node/src/main/java/com/hedera/services/ServicesState.java#L167) that ensures [accounts `0.0.800` and `0.0.801` exist](https://github.com/hashgraph/hedera-services/blob/03015-M-ImmutableSystemAccounts/hedera-node/src/main/java/com/hedera/services/state/migration/ReleaseTwentyFourMigration.java#L36), both with an [empty (immutable) key and a 1000 year expiration](https://github.com/hashgraph/hedera-services/blob/03015-M-ImmutableSystemAccounts/hedera-node/src/main/java/com/hedera/services/state/initialization/BackedSystemAccountsCreator.java#L122).
    * Streams synthetic `CryptoCreate` transactions and records for these two accounts.
- Adds a new `IMMUTABLE_ACCOUNT` [`KeyOrderingFailure` mode](https://github.com/hashgraph/hedera-services/blob/03015-M-ImmutableSystemAccounts/hedera-node/src/main/java/com/hedera/services/sigs/order/KeyOrderingFailure.java#L35) and [returns it](https://github.com/hashgraph/hedera-services/blob/03015-M-ImmutableSystemAccounts/hedera-node/src/main/java/com/hedera/services/sigs/metadata/StateChildrenSigMetadataLookup.java#L236) from `StateChildrenSigMetaLookup` for an account with an empty key.
    * Renames `MISSING_AUTORENEW_ACCOUNT` to `INVALID_AUTORENEW_ACCOUNT` and `MISSING_FEE_COLLECTOR_RESULT` to `INVALID_FEE_COLLECTOR_RESULT` to better reflect the actual HAPI response codes (which do not distinguish between missing and invalid accounts).
- Removes the assumption in `SigRequirements` that a key lookup can only fail because an account is missing; and now returns `KeyOrderingFailure` instead of `boolean` from internal helper methods [such as `addAccount()`](https://github.com/hashgraph/hedera-services/blob/03015-M-ImmutableSystemAccounts/hedera-node/src/main/java/com/hedera/services/sigs/order/SigRequirements.java#L904).
    * Overrides the `IMMUTABLE_ACCOUNT` failure mode in only [one place](https://github.com/hashgraph/hedera-services/blob/03015-M-ImmutableSystemAccounts/hedera-node/src/main/java/com/hedera/services/sigs/order/SigRequirements.java#L1187)---when an account is receiving funds in a `CryptoTransfer`. (And because immutable accounts cannot be associated to tokens, these funds will be exclusively ℏ.)
- Removes the questionable (dangerous?) `TerminalSigStatuses` class and fails immediately in [`NonPayerKeysScreen`](https://github.com/hashgraph/hedera-services/blob/03015-M-ImmutableSystemAccounts/hedera-node/src/main/java/com/hedera/services/state/logic/NonPayerKeysScreen.java#L54) for any non-`OK` status.
- Adds [an EET](https://github.com/hashgraph/hedera-services/blob/ea7f79793f3d5ea151bc664dd60ced6dfe0049b8/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java#L444) to verify immutable accounts can't be used for anything other than receiving ℏ.

**Related issue(s)**:

- Closes #3015

**Notes for reviewer**:
- In a future release we can add HAPI response codes that will let us surface _how_ an immutable account caused signature verification to fail; for example, `AUTORENEW_ACCOUNT_CANNOT_BE_IMMUTABLE`, `TREASURY_CANNOT_BE_IMMUTABLE`, and so on.

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/ServicesState.java [hunk 1]
```diff
@@ -161,9 +164,13 @@ public void addDeserializedChildren(List<MerkleNode> children, int version) {
...
 		int deserializedVersionFromState = getDeserializedVersion();
-		if (deserializedVersionFromState < RELEASE_0220_VERSION) {
-			blobMigrator.migrateFromBinaryObjectStore(this, deserializedVersionFromState);
-			init(getPlatformForDeferredInit(), getAddressBookForDeferredInit(), getDualStateForDeferredInit());
+		if (deserializedVersionFromState < RELEASE_0240_VERSION) {
+			stakeFundingMigrator.accept(this);
+			if (deserializedVersionFromState < RELEASE_0220_VERSION) {
+				// Remove after confirming no regression tests use 0.21.x states
+				blobMigrator.migrateFromBinaryObjectStore(this, deserializedVersionFromState);
+				init(getPlatformForDeferredInit(), getAddressBookForDeferredInit(), getDualStateForDeferredInit());
```
Comment (Neeharika-Sompalli) [inline hedera-node/src/main/java/com/hedera/services/ServicesState.java:172]: I guess we can remove these and make sure regression is updated to latest state ?

Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/ServicesState.java:172]: If there are no objections, I will remove! Mirror node team has asked for an enhancement to stream synthetic `CryptoCreate` records for these two accounts, so I need to push again anyways. 😁

**Review by Neeharika-Sompalli — DISMISSED** [review]
minor questions, otherwise LGTM

File: hedera-node/src/main/java/com/hedera/services/ServicesState.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/src/main/java/com/hedera/services/ServicesState.java:172]: Sounds good .. 😄

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM !

**Pull request merged**